New Features

SandyGifford opened the pull request.

SandyGifford committed:
added Game.play(...) function

- Added Game.play(string) function
- Takes a string, the first word is interpreted as the function to use,
the rest as arguments
- Keeps player from seeing all available options if console has
type-ahead
- Allows for more dynamic actions
- Allows for closer to "as you would say it" commands

SandyGifford committed:
NewGame function

added NewGame function.  Creates a Game object; does not return object,
instead returns a function which takes one parameter "input" which is
then passed to that Game object's Game.play() function.  This does two
things:

1)  Entirely hides the Game object from the end user
2)  Allows the end user to enter commands with a single function, no
"dots"

SandyGifford committed:
Simplification, usability

- Simplified NewGame function to simply return game.play instead of a
function wrapping around game.play
- Commands can now be entered in any case
- Removed player object, all functions moved to private member obscuring
them from end user
- Play function now returns a series of dashes, suppressing "undefined"
text in console and replacing it with an easily identifiable line
between commands

SandyGifford committed:
Updated Readme

- Updated README.md
- Fixed indenting

mimiflynn commented on public/js/westward-bound.js:
```diff
@@ -1,68 +1,65 @@
+			message: 'Welcome to your new life in the Rockies. \nMake wise decisions. \n"Live long and prosper"'
+		}
+	]
+});
```
Gotta have that new line ;)

SandyGifford replied:
Agree on Node.js - didn't realize you had kept it.  Also, learning so much!

mimiflynn commented on README.md:
````diff
@@ -1,21 +1,51 @@
-npm start
-```
-
-Open your browser to `http://localhost:1337'
````
I feel like this part of the `README.md` might still be necessary in case someone wants to work on it and add npm dependencies and such.

mimiflynn commented on README.md:
```diff
@@ -1,21 +1,51 @@
+			message: 'Welcome to your new life in the Rockies. \nMake wise decisions. \n"Live long and prosper"' // End game message
+		}
+	]
+};
```
Probably should wrap this code block in ```. Refer to the [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/) docs to see codeblock formatting and other options.

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,124 @@
-  };
+function NewGame(name, options)
+{
+	var game = new Game(name, options);
```
Guess we'll have to define 2 space or 4 space for this project... I had been doing 2 spaces. Also, it makes it difficult for me to completely see what the full diff is since git is saying its all new ;)

SandyGifford committed:
Newly organized game structure

Games now have a greater level of complexity allowing for:
- Custom actions
- Multiple paths
- Saving of variables game wide, or just to a certain frame (room)

Additionally, game is slightly more verbose.

SandyGifford committed:
Ability to "use" items

- Items can now be used on objects.  Example game contains a key that
can be used on a certain door.
- Expanded overall game.
- More functions accessible to frame functions
- Fixed several cases where missing members from a frame would cause
errors
- Pronoun strings allow items to specify how to refer to them (ie: "a
key", "an orange", "some hummus")
- Can check inventory with "inventory" command
- Support for one-word commands

SandyGifford committed:
Inventory functions completed, end game functions

- Finished inventory functions (add, remove, has, get)
- Functions for winning and losing disable game, print message
- Fleshed out example game, complete with treasure chest at the end!
- Made some functions more flexible

Commit:
Update README.md

Blocked out code

mimiflynn commented on README.md:
````diff
@@ -1,22 +1,54 @@
+Structure of the options object:
 
 ```
-npm start
````
Please add this part back, probably at the end of the file. Its a simple project right now, but eventually we will have a build process that will make `npm install` and `npm start` seem more relevent.

mimiflynn commented on README.md:
````diff
@@ -1,22 +1,54 @@
 ```
 
-Open your browser to `http://localhost:1337'
-
````
Be sure not to leave this part out.

SandyGifford committed:
Fixed small syntax.  First draft of readme.

- Readme is actually useful now.  Covers, playing, and writing using
API.  Included Node.js info
- Alphabetized functions (silly, need to reorganize again later)
- Fixed addItemToFrame function
- Fixed some capitalization
- Fixed small behavior involving doors in example game

SandyGifford committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	README.md

SandyGifford committed:
Fixed example in readme

Readme had an incomplete example

SandyGifford committed:
Updated README for consistency

Code highlights and terminology changes.

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,300 @@
-}
+(function()
+{
+	var div = "--------------------------------------------";
```
Single quotes and 2 space tabs preferred.

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,300 @@
+	var div = "--------------------------------------------";
+	
+	Game = function(name, options)
+	{
```
More regarding code style:

```
function (arguments) {
  var things = 'stuff';

  if (things === 'stuff') { // three equals for type checking and curly braces for clarity
    // do things
  }
}
```

SandyGifford replied:
Curly braces on same line?  Or are you saying  no single-lined if-statements?

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,300 @@
+		var cFrameName   = ""   ;
+		
+		if(!options.frames.entry)
+			throw "Game requires that exactly one frame be named \"entry\"";
```
Nice!

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,300 @@
+		{
+			gameActive = false;
+			
+			if(message)
```
Love the custom message option.

mimiflynn commented on public/js/console-game.js:
```diff
@@ -1,76 +1,300 @@
+					return func;
+				else if(func)
+					return func.apply(g);
+				else
```
Totes prrf.

mimiflynn commented on public/js/complex-game.js:
```diff
@@ -0,0 +1,273 @@
+var g = NewGame("Complex Game", {
+	version        : "0.1",
+	intro          : "Oh look!  A house!  Let's go in it!",
```
Think we should, by default, prepend game function directions? or do we just want people to dig through the window, aka hard mode?

PacmanBits commented:
All available functions?   Or just the basic ones like "move"

mimiflynn commented:
Yeah, basic functions that a player needs to know about... maybe even a help function. I could make it after this is merged, just a thought. Be nice if we had a better place for a conversation about future features.

SandyGifford committed:
Style Guidelines and File Management

- Brought code styling closer to original guidelines.
- Removed "westward-bound.js" (no longer compatible with current model)
- Renamed "complex-game".  Now called "The Mansion".  Changed in game
name, file name to be changed shortly.

Commit:
Changed File Name

Changed file name to match game name

Commit:
Changed script pointer for game code

Changed script pointer for game code (The-Mansion.js, previously complex-game.js)

PacmanBits commented:
I like the idea of a help function that gives general tips.  Def a good idea.  Maybe something that an API user could override, too (in addition to our basic tips).

I've been meaning to open a wiki for this rep, I'll see if I can get around to it later today.

SandyGifford committed:
Corrected indentation

Changed 4-space indents to 2
